nomos-da/network/core/src/protocols/sampling/requests/request_behaviour.rs

```diff
@@ -359,6 +364,24 @@ where
         sample_response: SampleResponse,
         stream: SampleStream,
     ) -> Poll<ToSwarm<<Self as NetworkBehaviour>::ToSwarm, THandlerInEvent<Self>>> {
+        match &sample_response {
+            SampleResponse::Share(_) | SampleResponse::Commitments(_) => {
+                self.opinion_events.push_back(OpinionEvent::Positive {
+                    peer_id,
+                    session_id: None, // None for current session
```

Comment:
Could this be SessionType instead of an Option? We could get the current session number from the membership, right?

Reply:
I think it is important for historic to see which session it is as we should probably discard older session then S-1. But the regular behavior works on the latest session anyway, right? and the da-network will know which session it is when it receives events as it will know current session (it is subscribed to membership)

Reply:
> da-network will know which session it is when it receives events

If we make membership handler know the session number, then da network wouldn't need to care about for which session opinions came from, behaviour could use the session number from membership handler and set it instead of None, wdyt?